### DISCUSSION TITLE
Focusing MUI text fields

### DISCUSSION THREAD

**Discussion by davidsl4** [react-hook-form/react-hook-form#8719] [Q&A]
Hi, I am trying to make MUI text fields focus once the form is submitted but failed by RHF rules.
As far as I understand, I need to attach the element to the `ref` property returned by the `useController` hook, which I did. But for some reason, the `ref` value in RHF model is set to `{ name: 'myInputsName' }.
I did some debugging; it looks like the `ref` was initially set correctly; it has all the functions needed to make focusing available. But once it is mounted and rendered, its value drops back to just the input's name.

Here is my code
<img width="274" alt="image" src="https://user-images.githubusercontent.com/21009377/180481239-c49bc281-1521-41e3-9acf-e2e120a604d6.png">
<img width="574" alt="image" src="https://user-images.githubusercontent.com/21009377/180481348-801f5c6b-50ed-46e4-a89b-4dea7c3698f8.png">

I really appreciate any help you can provide.

**Comment by Moshyfawn** [comment]
You'll have to provide a minimal repro codesandbox in order for us to debug your issue
